### PR TITLE
FTX: Various bug fixes

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

Fixes various bugs in FTX;

1) JSON tag value and variable type mismatch.
2) Assumption of a URL encoded request not allowing for optional requests, which throws an error on authenticated endpoints. 
3) Wrong HTTP method for submitting margin lending offers.
4) Implement limit checks for tests
5) Improve test error checking
6) Usage of timestamps which are relevant to the market being tested
7) Mismatch between public/private endpoints
8) Allow for case insensitivity

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run
- [x] authenticated requests

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes
- [N/A] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/ftx/ftx_test.go [hunk 1]
```diff
@@ -312,8 +330,7 @@ func TestSubmitLendingOffer(t *testing.T) {
 	if !areTestAPIKeysSet() || !canManipulateRealOrders {
 		t.Skip()
 	}
-	_, err := f.SubmitLendingOffer("btc", 0.1, 500)
-	if err != nil {
+	if err := f.SubmitLendingOffer("btc", 0.1, 500); err != nil {
```
Comment (gloriousCode) [inline exchanges/ftx/ftx_test.go:333]: "btc" as a currency does not exist, you could use "BTC" or something else

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Nice changes! One little nit, but otherwise, everything works 🎉

**Review by gloriousCode — APPROVED** [review]
Thanks for making those changes and improving case SENSITIVITY issues. tACK!

File: exchanges/ftx/ftx_types.go [hunk 2]
```diff
@@ -14,6 +14,12 @@ type MarginFundingData struct {
 	Previous float64 `json:"previous"`
 }
 
+// MarginDailyBorrowStats stores the daily borrwed amounts
```
Comment (shazbert) [inline exchanges/ftx/ftx_types.go:17]: borrwed -> borrowederino

File: exchanges/ftx/ftx_test.go [hunk 3]
```diff
@@ -91,11 +91,17 @@ func TestGetOrderbook(t *testing.T) {
 
 func TestGetTrades(t *testing.T) {
 	t.Parallel()
-	_, err := f.GetTrades(spotPair, 0, 0, 200)
+	// test empty market
+	_, err := f.GetTrades("", 0, 0, 200)
+	if err == nil {
+		t.Error(err)
```
Comment (shazbert) [inline exchanges/ftx/ftx_test.go:97]: err --> "error cannot be nil"

File: exchanges/ftx/ftx_test.go [hunk 4]
```diff
@@ -91,11 +91,17 @@ func TestGetOrderbook(t *testing.T) {
...
+	}
+	// no limit or time vals
+	_, err = f.GetTrades(spotPair, 0, 0, 0)
 	if err != nil {
 		t.Error(err)
 	}
-	_, err = f.GetTrades(spotPair, 0, 0, 5)
+	_, err = f.GetTrades(spotPair, 1559881511, 1559901511, 5)
```
Comment (shazbert) [inline exchanges/ftx/ftx_test.go:104]: Just a note: This returns no data as this is before FTT traded on FTX so it doesn't really test the limit.

**Review by shazbert — CHANGES_REQUESTED** [review]
I tested endpoints and they are good to go. Just timestamp issues are throughout in testing but not a big deal.

File: exchanges/ftx/ftx_test.go (hunk 4 above)
Comment (thrasher-) [reply exchanges/ftx/ftx_test.go]: Good catch! Updated these on the public endpoint tests and made them configurable for the auth ones

File: exchanges/ftx/ftx_test.go (hunk 3 above)
Comment (thrasher-) [reply exchanges/ftx/ftx_test.go]: There were quite a lot of these, they should be fine now

File: exchanges/ftx/ftx_test.go [hunk 5]
```diff
@@ -91,33 +96,70 @@ func TestGetOrderbook(t *testing.T) {
 
 func TestGetTrades(t *testing.T) {
 	t.Parallel()
-	_, err := f.GetTrades(spotPair, 0, 0, 200)
+	// test empty market
+	_, err := f.GetTrades("", 0, 0, 200)
+	if err == nil {
+		t.Error("empty market should return an error")
+	}
+	// no limit or time vals
```
Comment (MadCozBadd) [inline exchanges/ftx/ftx_test.go]: Comment should be shifted down a few lines

**Review by MadCozBadd — COMMENTED** [review]
Great changes! No real issues found except for a misplaced comment

**Review by MadCozBadd — APPROVED** [review]
Nevermind then

**Review by shazbert — APPROVED** [review]
Thank you for fixing the nits and adding the correct associated timestamps. tACK!

File: exchanges/ftx/ftx_test.go (hunk 3 above)
Comment (gloriousCode) [reply exchanges/ftx/ftx_test.go]: It's unrelated to FTX, but there are two more instances after I ctrl+f'd, one in btcmarkets_test.go:247 and another in bitfinex_test.go:405 if you feel like completely eradicating this scourge.

**Review by gloriousCode — APPROVED** [review]
I appreciate the more thorough tests. tACK!
# 🌟

**Pull request merged**